test.py

```diff
@@ -118,3 +120,39 @@ def test_nested_input():
 def test_function_mixup():
     """Make sure that different functions with the same args are cached under different keys"""
     assert get_random_with_args(1) != get_another_random_with_args(1)
+
+
+def callTextChangeTest():
+    process = Popen(["python", "textChangeTest.py"], stdout=PIPE)
+    (output, err) = process.communicate()
+    exit_code = process.wait()
+    assert exit_code == 0
+
+    return output.strip()
+
+
+def changeText(orig, new):
+    with open("textChangeTest.py", "r+") as f:
+        text = f.read()
+        text = re.sub(orig, new, text)
```

Comment:
I'd simplify this with another with statement for writing the text.

```suggestion
filename = "textChangeTest.py"
with open(filename, 'w') as f:
    f.write(re.sub(orig, new, text))
```